python/lsst/pex/config/config.py

```diff
@@ -319,6 +339,13 @@ class Field:
     container type (like a `lsst.pex.config.List`) depending on the field's
     type. See the example, below.
 
+    Fields can be annotated with a type similar to other python classes (see
+    `here <https://peps.python.org/pep-0484/#generics>`_ ). Unlike most other
+    uses in python, this has an effect at type checking *and* runtime. If the
+    type is specified with a class annotation, it will be used as the value of
+    the ``dtype`` in the ``Field`` and there is no need to specify it as an
+    argument during instantiation.
```

Comment:
A really simple example of this syntax in the section below, and a forward-reference to that here, would be very useful.  You could just add another `Field` to the config class in the existing example.